sdk/core/azure-core-tracing-opentelemetry/src/main/java/com/azure/core/tracing/opentelemetry/OpenTelemetryTracerProvider.java

```diff
@@ -29,12 +30,32 @@ public OpenTelemetryTracerProvider() {
      * @param libraryName Azure client library package name
      * @param libraryVersion Azure client library version
      * @param azNamespace Azure Resource Provider namespace.
-     * @param options instance of {@link com.azure.core.util.TracingOptions}
+     * @param applicationOptions instance of {@link com.azure.core.util.TracingOptions} passed by the application.
      * @return a tracer instance.
      */
     @Override
-    public Tracer createTracer(String libraryName, String libraryVersion, String azNamespace, TracingOptions options) {
+    public Tracer createTracer(String libraryName, String libraryVersion, String azNamespace,
+        TracingOptions applicationOptions) {
         Objects.requireNonNull(libraryName, "'libraryName' cannot be null.");
-        return new OpenTelemetryTracer(libraryName, libraryVersion, azNamespace, options);
+
+        final LibraryTelemetryOptions libraryOptions = new LibraryTelemetryOptions().setLibraryName(libraryName)
+            .setLibraryVersion(libraryVersion)
+            .setResourceProviderNamespace(azNamespace);
+        return new OpenTelemetryTracer(libraryOptions, applicationOptions);
+    }
+
+    /**
+     * Creates OpenTelemetry-based implementation of {@link Tracer}
+     *
+     * @param libraryOptions Library-specific telemetry options.
+     * @param applicationOptions Tracing options configured by the application.
+     * @return a tracer instance.
+     */
+    @Override
+    public Tracer createTracer(LibraryTelemetryOptions libraryOptions, TracingOptions applicationOptions) {
+        Objects.requireNonNull(libraryOptions, "'libraryOptions' cannot be null.");
+        Objects.requireNonNull(libraryOptions.getLibraryName(), "'libraryOptions.getLibraryName()' cannot be null.");
```

Comment:
Is `libraryName` a required param for all? If so, we should make this is a constructor param in `LibraryTelemetryOptions`

Reply:
right!